### PR TITLE
redis@5 5.0.9 (new formula)

### PR DESCRIPTION
**Pull request by dtrodrigues**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by bayandin** [comment]
It should be just `redis@5` (not `redis@5.0`).

And I have to ask, does it meet acceptable versioned formula criteria https://docs.brew.sh/Versions#acceptable-versioned-formulae ?

**Comment by chenrui333** [comment]
@dtrodrigues, like @bayandin said, please use `redis@5.rb` instead of `redis@5.0.rb`

File: Formula/redis@5.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+class RedisAT50 < Formula
```
Comment (SMillerDev) [inline Formula/redis@5.rb:1]:
```suggestion
class RedisAT5 < Formula
```

**Comment by Bo98** [comment]
If we are going to use `redis@5` we should rename the `redis@6.0` alias to `redis@6`, and potentially rename `redis@4.0` to `redis@4` (with a `formula_renames.json` entry).

**Comment by dtrodrigues** [comment]
I had initially used `redis@5.0.rb` since the other two versioned formulae included the minor version (`redis@4.0.rb` and `redis@3.2.rb`). I don't have a preference either way but I agree it should be consistent.

**Comment by bayandin** [comment]
I found this doc regarding supporting older versions by redis https://redis.io/topics/releases and it states:

> Older versions are not supported as we try very hard to make the Redis API mostly backward compatible. Upgrading to newer versions is usually trivial.
For example, if the current stable release is 2.6.x, we accept bug reports and provide support for the previous stable release (2.4.x), but not for older ones such as 2.2.x.
When 2.8 becomes the current stable release, the 2.6.x will be the oldest supported release.


And it raises 2 questions in my mind:
- should we drop all the redis versions older than 5?
- do we really need to have a separate formula for previous version (5 for this case)? Does it have breaking changes?

**Comment by Bo98** [comment]
Original motivation for `redis@3.2` (#15775): "4.0.0 is not quite stable yet and not mainstream for development"

Original motivation for `redis@4.0` (#33305): "we cannot do redis@4". Took 2 months to merge and had little noise in that time.

With that said, they do end up being relatively highly used: `redis@3.2` still has 476 installs in the last 30 days while `redis@4.0` had 418.

**Comment by SMillerDev** [comment]
I saw that statement on the redis page as well but 3.2 did have a pretty recent release as far as I can tell.

**Comment by Bo98** [comment]
Yeah 3.2 got a release after 5.0.

**Comment by dtrodrigues** [comment]
As @Bo98 pointed out, the motivation for the 3.2 formula was lack of stability in 4.0 and the motivation for the 4.0 formula was completeness. I only added the 5.0 formula for completeness as well, not due to a need.

In response to @bayandin, Redis 6 _shouldn't_ have any breaking changes so it probably makes sense to get rid of the 5.0 versioned formula as well unless/until the need arises.

**Comment by Bo98** [comment]
There's some formulae which I will never understand _why_ people want versioned formulae for - just that they do. Redis is one of them.

**Comment by bayandin** [comment]
Thanks @dtrodrigues for trying and for understanding.

I'm closing this PR now, we will return to it if required

**Comment by dtrodrigues** [comment]
Should `redis@3.2.rb` and `redis@4.0.rb` be removed in this case? I see `redis@3.2.rb` being marked for cleanup in #52141 but given this discussion, both should probably be removed especially if we aren't going to keep `redis@5`.

**Comment by Bo98** [comment]
We now have the ability to deprecate formulae, so we could do that.

**Pull request closed without merging**